oml/metrics/embeddings.py

```diff
@@ -199,27 +201,32 @@ def compute_metrics(self) -> TMetricsDict_ByLabels:  # type: ignore
 
         self._calc_matrices()
 
-        args_retrieval_metrics = {
-            "cmc_top_k": self.cmc_top_k,
-            "precision_top_k": self.precision_top_k,
-            "map_top_k": self.map_top_k,
-            "fmr_vals": self.fmr_vals,
-        }
-        args_topological_metrics = {"pcf_variance": self.pcf_variance}
+        # todo 522: temp solution
+        max_k_arg = max([*self.cmc_top_k, *self.precision_top_k, *self.map_top_k])
+        k = min(self.distance_matrix.shape[1], max_k_arg)  # type: ignore
+        _, retrieved_ids = torch.topk(self.distance_matrix, largest=False, k=k)
+        gt_ids = [torch.nonzero(row, as_tuple=True)[0].tolist() for row in self.mask_gt]  # type: ignore
```

Comment:
Why retrieve `gt_ids` as `List[List[int]]` instead of `List[Tensor]`? I propose updating the `calc_retrieval_metrics` function to accept `gt_ids` of type `List[Tensor]` to eliminate the need for double conversion between tensor and list.

Reply:
agree